### ISSUE TITLE
All containers spontaneously stop randomly / intermittently

### ISSUE THREAD

**Issue by onmishkin** [docker/for-mac#396]
### Expected behavior

Containers should not stop unless they exit or I stop them.

### Actual behavior

Every day or so I notice all my containers have stopped.

### Information

Diagnostic ID: FF7D4587-F19D-49D1-96B0-EBC7133A7A18
Docker for Mac: 1.12.0-a (Build 11213)
macOS: Version 10.11.6 (Build 15G31)
[OK]     docker-cli
[OK]     app
[OK]     moby-syslog
[OK]     disk
[OK]     virtualization
[OK]     system
[OK]     menubar
[OK]     osxfs
[OK]     db
[OK]     slirp
[OK]     moby-console
[OK]     logs
[OK]     vmnetd
[OK]     env
[OK]     moby
[OK]     driver.amd64-linux

### Steps to reproduce

No known repro steps yet. I'm attaching a relevant excerpt from the logs around the time of the failure. Is the VM dying and being restarted?

[Docker.log.zip](https://github.com/docker/for-mac/files/431170/Docker.log.zip)

**Comment by emhagman** [comment]
This happens to me as well at least two times a day. Doesn't not appear to follow a pattern at the moment.

**Comment by jamesdixon** [comment]
@onmishkin @emhagman this happens to me as well, but it's only after my mac sleeps and then is awoken. Have you experienced this?

**Comment by onmishkin** [comment]
@jamesdixon I do put my Mac to sleep periodically but I'll have to start paying attention to see whether that correlates with the container death issue. My gut says no, but we'll see.

**Comment by jamesdixon** [comment]
@onmishkin keep us updated. There is an open issue for the problem I spoke of: #85

**Comment by samoht** [comment]
Detected symptom of problem 'invariant-violated' in FF7D4587-F19D-49D1-96B0-EBC7133A7A18/20160822-164322.

An invariant within the code of one of the components was not held. Docker needs to be restarted.

May be related to docker/for-mac#89

The following log matches:
```
Aug 15 16:13:37 Nats-MacBook-Pro-22 Docker[com.docker.driver.amd64-linux][554] <Notice>: VM: 2016-08-15 20:13:37 +0000 UTC daemon.info vsudd: 52 Done. read: 279 written: 95
Aug 15 16:13:37 Nats-MacBook-Pro-22 Docker[com.docker.driver.amd64-linux][554] <Notice>: VM: 2016-08-15 20:13:37 +0000 UTC daemon.info vsudd: 50 Done. read: 279 written: 95
Aug 15 16:13:37 Nats-MacBook-Pro-22 Docker[com.docker.osxfs][550] <Notice>: Volume.stop ca03497f66eeb43c16fa6bf970e9bc4f19343c550caa4232a0fdc4fac20af07e (paths = [/Users/nmishkin/docker/myenv3/elasticsearch/data])
Aug 15 16:13:37 Nats-MacBook-Pro-22 Docker[com.docker.osxfs][550] <Error>: WATCH TREE REMOVAL EXISTENCE INVARIANT VIOLATED
Aug 15 16:13:37 Nats-MacBook-Pro-22 Docker[com.docker.driver.amd64-linux][554] <Notice>: VM: 2016-08-15 20:13:37 +0000 UTC kern.info kernel: br-3ef21c766c8f: port 3(veth3c84d16) entered disabled state
Aug 15 16:13:37 Nats-MacBook-Pro-22 Docker[com.docker.driver.amd64-linux][554] <Notice>: VM: 2016-08-15 20:13:37 +0000 UTC kern.info kernel: veth0db88f5: renamed from eth0
Aug 15 16:13:37 Nats-MacBook-Pro-22 Docker[com.docker.driver.amd64-linux][554] <Notice>: VM: 2016-08-15 20:13:37 +0000 UTC kern.info kernel: br-3ef21c766c8f: port 3(veth3c84d16) entered disabled state
```

We also advised to upgrade to the latest stable (1.12.1) or latest beta (1.12.2-rc1-beta27) where this bug is fixed. /cc @dsheets

**Comment by dsheets** [comment]
This invariant violation was a symptom of a use-after-free bug in the file system event watching code which occurred due to multiple containers with volume mounts racing to set up their mounts. It is possible that it caused your hangs or crashes and has been fixed in the recently released Beta 29.3. Please open a new issue for any further problems. Thanks!
